### PR TITLE
Filter measurements search by focal and observed species

### PR DESCRIPTION
**Pull request by mauberti-bc**
## Links to Jira Tickets

- [SIMSBIOHUB-557](https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-557)

## Description of Changes

- New endpoint for retrieving species observed in a survey
- Allows measurement search to only return measurements that focal species/observed species can have
- Updates styling of the configure observations table dialog

## Testing Notes

- Requires Critterbase changes: https://github.com/bcgov/critterbase-api/pull/63
- Requires BioHub changes: https://github.com/bcgov/biohubbc-platform/pull/250

### CODE REVIEW

File: app/src/features/surveys/observations/observations-table/configure-columns/components/measurements/search/MeasurementsSearch.tsx [hunk 1]
```diff
@@ -25,20 +35,58 @@ export interface IMeasurementsSearchProps {
...
+    critterbaseApi.xref.getMeasurementTypeDefinitionsBySearchTerm(searchTerm, tsns)
+  );
+
+  const hierarchyDataLoader = useDataLoader((tsns: number[]) => biohubApi.taxonomy.getTaxonHierarchyByTSNs(tsns));
+
+  useEffect(() => {
+    if (!observationsContext.observedSpeciesDataLoader.data) {
+      observationsContext.observedSpeciesDataLoader.load();
+    }
+  }, [observationsContext.observedSpeciesDataLoader.data]);
```
Comment (MacDeluca) [inline app/src/features/surveys/observations/observations-table/configure-columns/components/measurements/search/MeasurementsSearch.tsx]: Eslint ignore dependency warnings

File: app/src/features/surveys/observations/observations-table/configure-columns/components/measurements/search/MeasurementsSearch.tsx [hunk 2]
```diff
@@ -25,20 +35,58 @@ export interface IMeasurementsSearchProps {
...
+  const focalOrObservedSpecies: number[] = [
+    ...(surveyContext.surveyDataLoader.data?.surveyData.species.focal_species.map((species) => species.tsn) ?? []),
+    ...(observationsContext.observedSpeciesDataLoader.data?.map((species) => species.tsn) ?? [])
+  ];
+
+  useEffect(() => {
+    if (focalOrObservedSpecies.length) {
+      hierarchyDataLoader.load(focalOrObservedSpecies);
+    }
+  }, []);
```
Comment (MacDeluca) [inline app/src/features/surveys/observations/observations-table/configure-columns/components/measurements/search/MeasurementsSearch.tsx]: Update dependency array or ignore

File: app/src/hooks/cb_api/useXrefApi.tsx [hunk 3]
```diff
@@ -21,13 +22,21 @@ export const useXrefApi = (axios: AxiosInstance) => {
...
   const getMeasurementTypeDefinitionsBySearchTerm = async (
-    searchTerm: string
+    name: string,
+    tsns?: number[]
   ): Promise<CBMeasurementSearchByTermResponse> => {
-    const { data } = await axios.get(`/api/critterbase/xref/taxon-measurements/search?name=${searchTerm}`);
+    const t = tsns?.map((tsn) => Number(tsn));
+    const { data } = await axios.get(`/api/critterbase/xref/taxon-measurements/search`, {
+      params: { name, tsns: t },
+      paramsSerializer: (params) => {
```
Comment (MacQSL) [inline app/src/hooks/cb_api/useXrefApi.tsx:36]: Params serializer is included in the critterbase axios client config now

**Review by MacDeluca — APPROVED** [review]
SHIPIT 🚢

**Pull request merged**